build/geanylua.m4

```diff
@@ -4,17 +4,35 @@ AC_DEFUN([GP_CHECK_GEANYLUA],
 
     AC_ARG_WITH([lua-pkg],
         AC_HELP_STRING([--with-lua-pkg=ARG],
-            [name of Lua pkg-config script [[default=lua5.1]]]),
+            [name of Lua pkg-config script [[default=lua]]]),
         [LUA_PKG_NAME=${withval%.pc}],
-        [LUA_PKG_NAME=lua5.1
+        [LUA_PKG_NAME=lua
 
-        for L in lua5.1 lua51 lua-5.1 lua; do
+        for L in "$LUA_PKG_NAME" lua lua53 lua54 lua52 luajit lua51; do
```

Comment:
Why this order?  And also, why drop `lua5.1` and `lua-5.1` which probably were added for a reason?

Reply:
Seeing your other comment, I would think the "right" order would be `lua5.4 luajit lua5.3 lua5.2 lua5.1 lua`.

Not that as it is, `lua` is a bit problematic because if found, the version is only checked *after*, and so if at an unsupported version it breaks the test.

Reply:
Revised to remove Lua search.  User/packager is assumed to provide the correct name if default `lua` is not suitable.